cpu/cc3200/periph/gpio.c

```diff
@@ -0,0 +1,412 @@
+{
+    gpio(dev)->im &= ~(gpio_pin_mask(dev));
+}
+
+int gpio_init_int(gpio_t dev, gpio_mode_t mode, gpio_flank_t flank,
+                  gpio_cb_t cb, void *arg)
+{
+    /* Note: gpio_init() also checks if the gpio is enabled. */
+    int res = gpio_init(dev, mode);
+    if (res != 0) {
+        return res;
+    }
+
+    assert(flank != GPIO_NONE);
+
+    uint8_t portNum = gpio_port_num(dev);
+    uint8_t pinNum  = gpio_pin_num(dev);
+    uint8_t bit     = gpio_pin_mask(dev);
+    /* convert to number since we use the address for later compare */
+    uint32_t portBase = (uint32_t)gpio_port_by_num(portNum);
+
+    /* store callback information; */
+    isr_ctx[portNum][pinNum].cb  = cb;
+    isr_ctx[portNum][pinNum].arg = arg;
+
+    ROM_IntMasterDisable();
+
+    /* clear interrupt specified pin */
+    gpio(dev)->icr = bit;
+
+    /* configure active flanks */
+    gpio(dev)->ibe =
+            (flank & GPIO_BOTH) ? gpio(dev)->ibe | bit : gpio(dev)->ibe & ~bit;
+    gpio(dev)->is = (flank & GPIO_VALUE_LOW) ? gpio(dev)->is | bit :
+                                               gpio(dev)->is & ~bit;
+    gpio(dev)->iev = (flank & GPIO_RISING) ? gpio(dev)->iev | bit :
+                                             gpio(dev)->iev & ~bit;
+
+    /* enable gpio interripts */
+    gpio(dev)->im |= bit;
+
+    /* register interrupt handlers */
+    /* TODO: replace with cortex common */
+    switch (portBase) {
+    case GPIOA0_BASE:
+        ROM_GPIOIntRegister(portBase, isr_gpio_a0);
+        ROM_IntEnable(INT_GPIOA0);
+        break;
+    case GPIOA1_BASE:
+        ROM_GPIOIntRegister(portBase, isr_gpio_a1);
+        ROM_IntEnable(INT_GPIOA1);
+        break;
+    case GPIOA2_BASE:
+        ROM_GPIOIntRegister(portBase, isr_gpio_a2);
+        ROM_IntEnable(INT_GPIOA2);
+        break;
+    case GPIOA3_BASE:
+        ROM_GPIOIntRegister(portBase, isr_gpio_a3);
+        ROM_IntEnable(INT_GPIOA3);
+        break;
+    case GPIOA4_BASE:
+        ROM_GPIOIntRegister(portBase, isr_gpio_a4);
+        ROM_IntEnable(INT_GPIOA4);
+        break;
+    }
+
+    ROM_IntMasterEnable();
+
+    return 0;
+}
+#endif
+
+/**
+ * @brief gpio_write writes to a GPIO pin dev (external) the value value
+ *
+ * @param dev external pin
+ * @param value value to be written (will be masked)
+ */
+void gpio_write(gpio_t dev, int value)
+{
+    uint8_t port           = gpio_port_num(dev);
+    unsigned char ipin     = gpio_pin_mask(dev);
+    unsigned long portAddr = (unsigned long)gpio_port_by_num(port);
+    /* write to pin at portBase + pinOffset */
+    HWREG(portAddr + (ipin << 2)) =
+            _gpio_pin_value_mask(pin_to_gpio_num[gpio_pin_num(dev)], value);
+}
+
+/**
+ * @brief read a pins value
+ *
+ * @param dev external pin
+ * @return int current value of a pin
+ */
+int gpio_read(gpio_t dev)
+{
+    uint8_t port           = gpio_port_num(dev);
+    unsigned char ipin     = gpio_pin_mask(dev);
+    unsigned long portAddr = (unsigned long)gpio_port_by_num(port);
```

Comment:
Again, use the integer types with defined size for register variables.
It doesn't make any difference as by definition this code will only run on a single platform, but it's a question of style ;)